tests/validation/tests/v3_api/test_multi_cluster_app.py

```diff
@@ -147,14 +671,58 @@ def create_project_client(request):
 
     def fin():
         client_admin = get_admin_client()
-        client_admin.delete(project[p1.id]["project"])
-        client_admin.delete(project[p2.id]["project"])
+        client_admin.delete(p1, ns1, p_client1)
+        client_admin.delete(p2, ns2, p_client2)
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture(scope='module')
+def create_project_client2(request):
```

Comment:
This could be updated to use a for loop to create all the resources and save them off on `project_detail2`, doing the same thing three times makes it hard to make changes.